### PR TITLE
[EC-51] EVM System Operations - SELFDESTRUCT

### PR DESCRIPTION
**Pull request by whysoserious**
~~Depends on https://github.com/input-output-hk/etc-client/pull/64 .~~
Minor changes: 
- Set log level in tests to `WARN`

### CODE REVIEW

File: src/test/scala/io/iohk/ethereum/vm/OpCodeGasSpec.scala [hunk 1]
```diff
@@ -437,5 +440,45 @@ class OpCodeGasSpec extends FunSuite with OpCodeTesting with Matchers with Prope
     }
   }
 
+  test(SELFDESTRUCT) { op =>
+    val stateGen = getProgramStateGen(
+      stackGen = getStackGen(elems = 2)
+    )
+
+    // Sending refund to an account created in current transaction
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/vm/OpCodeGasSpec.scala]: Good test but the description is inaccurate. `in current transaction` could mean on message call prior to this self destruct operation, in which case we don't charge gas for new account. Just say `// Sending refund to a non-existent account`.

File: src/test/resources/logback-test.xml [hunk 2]
```diff
@@ -6,7 +6,7 @@
         </encoder>
     </appender>
 
-    <root level="DEBUG">
+    <root level="WARN">
```
Comment (rtkaczyk) [inline src/test/resources/logback-test.xml]: Sure about this?

**Comment by rtkaczyk** [comment]
@whysoserious have you addressed the subtleties in YP section 7.1? If not, please remember to add a task to investigate this.

File: .gitignore [hunk 3]
```diff
@@ -6,8 +6,8 @@ project/plugins/project/
...
 .ensime_cache/
 
-
 # intellij scala worksheet
 *.sc
+*.log
 
-*.log
\ No newline at end of file
+notes.org
```
Comment (adamsmo) [inline .gitignore]: what is that ;) ?

File: build.sbt [hunk 4]
```diff
@@ -40,4 +40,4 @@ scalacOptions := Seq("-unchecked", "-deprecation", "-feature")
 testOptions in Test += Tests.Argument("-oD")
 
 (scalastyleConfig in Test) := baseDirectory.value / "scalastyle-test-config.xml"
-scalastyleSources in Test ++= {(unmanagedSourceDirectories in Integration).value}
+scalastyleSources in Test ++= {(unmanagedSourceDirectories in Integration).value}
```
Comment (adamsmo) [inline build.sbt]: do we want to remove new line from end of the line?

File: src/test/resources/logback-test.xml (hunk 2 above)
Comment (whysoserious) [inline src/test/resources/logback-test.xml]: Yes, logs are flooding my console

**Comment by whysoserious** [comment]
> have you addressed the subtleties in YP section 7.1? If not, please remember to add a task to investigate this.

Yes.

File: src/test/scala/io/iohk/ethereum/vm/OpCodeFunSpec.scala [hunk 5]
```diff
@@ -267,7 +268,7 @@ class OpCodeFunSpec extends FunSuite with OpCodeTesting with Matchers with Prope
 
         val withinLimits =
           stateIn.env.blockHeader.number - blockHeaderNumber.toBigInt <= 256 &&
-          blockHeaderNumber.toBigInt < stateIn.env.blockHeader.number
+            blockHeaderNumber.toBigInt < stateIn.env.blockHeader.number
```
Comment (adamsmo) [inline src/test/scala/io/iohk/ethereum/vm/OpCodeFunSpec.scala]: unnecessary formatting change?

File: src/test/resources/logback-test.xml (hunk 2 above)
Comment (rtkaczyk) [inline src/test/resources/logback-test.xml]: I'm worried that we might miss something important when running tests in the CI.

File: src/test/scala/io/iohk/ethereum/vm/OpCodeGasSpec.scala [hunk 6]
```diff
@@ -444,5 +445,45 @@ class OpCodeGasSpec extends FunSuite with OpCodeTesting with Matchers with Prope
...
   }
 
+  test(SELFDESTRUCT) { op =>
+    val stateGen = getProgramStateGen(
+      stackGen = getStackGen(elems = 2)
+    )
+
+    // Sending refund to a non-existent account
+    forAll(stateGen) { stateIn =>
+      val (Seq(_, refundDW), _) = stateIn.stack.pop(2)
```
Comment (rtkaczyk) [inline src/test/scala/io/iohk/ethereum/vm/OpCodeGasSpec.scala]: `refundDW` -> `refund`? Likewise in the `OpCodeFunSpec`

File: src/test/resources/logback-test.xml (hunk 2 above)
Comment (whysoserious) [inline src/test/resources/logback-test.xml]: Ok, I'll remove that. That change is not worth time spent on discussing that.

**Comment by adamsmo** [comment]
LGTM but it is outdated

**Comment by whysoserious** [comment]
@adamsmo. @rtkaczyk updated
